### DISCUSSION TITLE
How to create a compiled SCSS lib that's publishable to npm

### DISCUSSION THREAD

**Discussion by Cabinet20** [nrwl/nx#7714] [General]
I have previously had angular libraries for styles, to share with other projects. I created this npm package using webpack.

I have a style library in our new Nx mono-repo and although it's fine for applications within the mono-repo, I still need to make it available outside. This, I tried to do by making it publishable but it appears from the schema in the angular.json file that you can't use `assets`, `styles` and probably most importantly `stylePreprocessorOptions`. I then looked into using `webpackConfig` within the project/architect/build/options area but this again isn't supported in the angular schema for `projectType`s of type `library`.

Does anyone have a steer how you go about doing this in Nx?

Sorry if this is obvious and I'm missing something daft, but I am honestly a little stuck just now. Thanks!

**Comment by asherccohen** [comment]
I'm also trying to fix this. Any luck so far?

**Comment by caenguidanos** [comment]
You can't do something like [this](https://github.com/caenguidanos/nx-lib-starter/blob/main/packages/ui-styles/breakpoints/package.json)?
